core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java

```diff
@@ -200,6 +204,7 @@ public QueryStringQueryBuilder(StreamInput in) throws IOException {
         timeZone = in.readOptionalTimeZone();
         escape = in.readBoolean();
         maxDeterminizedStates = in.readVInt();
+        splitOnWhitespace = in.readBoolean();
```

Comment:
actually there should protection on 6.x too since we have not given up the idea about multi-version clusters yet